doc/reported_states_during_deployment.md

```diff
@@ -0,0 +1,353 @@
+                        "imageHash": "sha256:85fdb1e9675c837c18b75f103be6f156587d1058eced1fc508cdb84a722e4f82",
+                        "createOptions": "{}"
+                    },
+                    "env": {}
+                }
+            },
+            "$version": 196
+        }
+    }
+}
+```
+
+| Relevant Field | Meaning |
+|------------------------------------------|-------------------------------------------------------|
+| .properties.reported.lastDesiredVerison | The last version number received and acted on. |
+| .properties.reported.lastDesiredStatus.code | The current status of the deployment (see [table below](#last-desired-status-codes)).   |
+| .properties.reported.lastDesiredStatus.description | Text describing the deployment status. |
+| .properties.reported.systemModules.edgeAgent.runtimeStatus | See [module status table below](#module-and-system-module-runtime-status). |
+| .properties.reported.systemModules.edgeAgent.settings.image | Current image being run by container runtime. | 
+| .properties.reported.systemModules.edgeHub.runtimeStatus | See [module status table below](#module-and-system-module-runtime-status). |
+| .properties.reported.systemModules.edgeHub.settings.image | Current image being run by container runtime. |
+| .properties.reported.modules.*\<module\>*.runtimeStatus | See [module status table below](#module-and-system-module-runtime-status). |
+| .properties.reported.modules.*\<module\>*.settings.image | Current image being run by container runtime.  The image names are read from the container itself, so will reflect the currently running image. |
+
+#### Last Desired Status codes
+
+| Code | Meaning |
+|------|---------|
+| 200  | Successful deployment |
+| 400  | ConfigFormatError - deployment was not able to be read by edgeAgent |
+| 406  | Unknown |
+| 412  | InvalidSchemaVersion - deployment schema version not compatible with edgeAgent |
+| 417  | ConfigEmptyError - no deployment has been send to IoT Edge |
+| 500  | Failed - something has gone wrong when applying a deployment |
+
+The code will be 406(Unknown) until \$edgeAgent is able to connect to the IoT Hub service, and 
+will be set to 406 when \$edgeAgent shuts down.
+
+Once online, the code may be 417(ConfigEmptyError) until \$edgeAgent receives a non-empty 
+deployment.
+
+Once the deployment is successfully realized, the code will be 200(Ok).
+
+If the deployment is not successfully realized, the code will either be 400, 412, 417 or 500, based on the
+reason why it failed.
+
+#### Module and system module runtime status
+
+A module's runtime status is the current status of the modules as interpreted by the IoT Edge 
+runtime from the container status. 
+
+| status  | Meaning |
+|---------|---------|
+| unknown | Module status is unknown. |
+| backoff | Module has had problems running, so \$edgeAgent has scheduled the module to restart at a future time.|
```

Comment:
Question: In the "backoff" status, is the module running? or has it been stopped?